app/models/hbx_enrollment.rb

```diff
@@ -602,17 +601,14 @@ def propagate_selection
   end
 
   def update_renewal_coverage
-    if is_shop?
-      if successor_benefit_package = sponsored_benefit_package.successor
-        successor_application = successor_benefit_package.benefit_application
-        passive_renewals_under(successor_application).each{|en| en.cancel_coverage! if en.may_cancel_coverage? }
-        if active_renewals_under(successor_application).blank?
-          if successor_application.coverage_renewable?
-            renew_benefit(successor_benefit_package)
-          end
-        end
-      end
-    end
+    return unless is_shop? && (successor_benefit_package = sponsored_benefit_package.successor)
+    successor_application = successor_benefit_package.benefit_application
+    passive_renewals_under(successor_application).each{|en| en.cancel_coverage! if en.may_cancel_coverage? }
+    renew_benefit(successor_benefit_package) if active_renewals_under(successor_application).blank? && successor_application.coverage_renewable? && non_inactive_transition?
+  end
+
```

Comment:
@ymhari @raghuramg Hi Ram, can I also get your feedback here.

Reply:
@ymhari @raghuramg didn't we have plans to get rid of the successor enrollment links?